src/engine/strat_engine/metadata.rs

```diff
@@ -697,7 +697,7 @@ mod mda {
                 let region_size = (MIN_MDA_SECTORS / 4usize).bytes() +
                                   Bytes(region_size_ext as u64);
                 let header = MDAHeader {
-                    last_updated: Timespec::new(sec, nsec),
+                    last_updated: Local.timestamp(sec, nsec as u32),
```

Comment:
Better to change ```nsec``` type in ```mda_header``` method signature to ```u32``` and make other changes as necessary (rather than cast here).